Project Management Process.md

```diff
@@ -0,0 +1,168 @@
+                    |  
+  Milestones        |     Milestone 1         Milestone A
+    Milestone 1     |       Goal 1              Goal A
+    Milestone 2     |       Goal 2            Milestone B
+    Milestone 3     |       Goal 3              Goal B
+    Milestone A     |     Milestone 2           Goal C
+    Milestone B     |       Goal 4              Goal D
+    Milestone C     |     Milestone 3      
+                    |       Goal 5      
+  Projects          |  
+    Project 1       |   Backlog             Backlog
+    Project 2       |     Goal 1              Goal A
+    Project 3       |     Goal 2              Goal B
+                    |     Goal 3              Goal C
+                    |     Goal 4              Goal D
+                    |     Goal 5
+```
+
+### Goals
+At the heart of each project are the goals. They are the basic unit of work. Goals can be anything that ***adds value to a release***. This includes issues reported by users, feature requests, bugs, Pull-Requests, general tasks such as refactoring, documentation or research. Goals are collected, maintained and tracked in the backlog by the project lead.
+
+A goal description includes the following:
+- Name / Summary
+  - Describes what the goal is on high level
+  - Example 1: "Reduce the size of the go-ipfs executable"
+  - Example 2: "js-ipfs should have API Documentation"
+  - TODO: what is a good name for a goal?
+- Description
+  - What the problem is and how this goal solves it
+- Type
+  - Describe the type of this goal: is it a bug? a new feature? something else?
+- Status
+  - What is the current status of this goal. Is it to-be-reviewed? In progress? Blocked? Done?
+- Dependencies (where applicable)
+- Priority (optional, priority can be defined in the backlog)
+- Any additional information that helps to reach this goal (optional)
+
+TODO: a clear example
+
+### Backlog
+Backlog is the collection of all goals in a project. It should be as comprehensive as possible, including everything that needs to happen or should happen in a particular project. The backlog indicates priority by order: the items that have the highest priority are at the top of the list. The backlog is owned and maintained by the project lead. As part of daily work, the backlog gets updated and a general grooming should happen on weekly basis.
+
+A backlog has the following information:
+- All goals of a project
+- Priority of goals in sorted order
+
+While the backlog is usually one big list of goals, it may sometimes become convoluted. In this case, the backlog list can be split in two: Backlog and Icebox. The icebox section lists goals that will probably not be worked on anytime soon, or are not ready to be worked on yet, whereas the backlog itself contains the goals that are ready or "approved". If split, together they are called "the backlog" and icebox is referenced only in its specific meaning. The split and maintenance of both lists are done the project lead.
```

Comment:
Updated:

***An example pipeline could have the following stages:***

```
  ╭─────────────────────────────────── Pipeline ──────────────────────────────────────╮
  │────── 1. ────│──── 2. ───│─── 3. ──│────── 4. ─────│─── 5. ───│──── 6. ───│── 7. ─│
  │ Stages                                                                            │

                                                ╭────────────╮
──> Future Work ──> Backlog ──> Ready ──> In Progress ──> Review ──> Release ──> Done ──>
         │             │          │             │            │ 
         ╰─────────────╯          │             │            │
                                  ╰───────── Blocked ────────╯
    Phases
  │───────── Backlog ────────│──────────── Development ───────────│──── Complete ─────│
  ╰───────────────────────────────────────────────────────────────────────────────────╯
```

1. Future Work - the goal is identified and being fleshed out. It is not ready to be carried out at this moment. 
2. Backlog - the goal is well defined but not planned to be carried out at this moment.
3. Ready - the goal is planned to be carried out, waiting to be worked on.
4. In Progress - the goal is actively being worked on, this moment.
5. Review - the development of the goal is done, but a review needs to happen before it can be included in a release.
6. Release - the goal is waiting to be inckuded in a an official release.
7. Done - the goal is completed. No further action is necessary.

* Blocked - the goal cannot proceed as something is preventing it to move down the pipeline.